org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/res/loader/DynamicResourceInfo.java

```diff
@@ -106,6 +108,31 @@ public static DynamicResourceInfo fromPath(HttpServletRequest req, String path)
     return new DynamicResourceInfo(uiSession, components.getAdapterId(), components.getFilename());
   }
 
+  /**
+   * @param uiSessionResolver
+   *          non-null UI session ID -> {@link IUiSession} resolver
+   * @param jsonAdapterResolver
+   *          non-null json adapter ID -> {@link IJsonAdapter} resolver
+   * @param path
+   *          decoded path (non url-encoded)
+   */
+  public static DynamicResourceInfo fromPath(Function<String/*uiSessionId*/, IUiSession> uiSessionResolver, Function<String/*id*/, IJsonAdapter> jsonAdapterResolver, String path) {
+    DynamicResourcePathComponents parts = DynamicResourcePathComponents.fromPath(path);
+    if (parts == null) {
+      return null;
+    }
+    // resolve session id and adapter id
+    IUiSession uiSession = Optional.ofNullable(parts.getUiSessionId()).map(uiSessionResolver).orElse(null);
+    IJsonAdapter jsonAdapter = Optional.ofNullable(parts.getAdapterId()).map(jsonAdapterResolver).orElse(null);
```

Comment:
call `jsonAdapterResolver` after validating the `uiSession`

Reply:
We pass in IUiSession directly now, removed resolver indirection.